### PR TITLE
docs/adr: Initial import of the runtime update ADR

### PR DESCRIPTION
**Pull request by Yawning**

### CODE REVIEW

File: docs/adr/0013-runtime-upgrades.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+## Implementation
+
+Runtime descriptor related changes:
+
+```golang
+// Runtime represents a runtime.
+type Runtime struct { // nolint: maligned
+	cbor.Versioned
+
```
Comment (kostko) [inline docs/adr/0013-runtime-upgrades.md:30]: Consider eliding fields that are not modified.

File: docs/adr/0013-runtime-upgrades.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+```
+
+The intended workflow here is to:
+
+- Deploy runtimes with the initial Deployment populated.
+
+- Update the runtime version via the deployment of a new version
+  of the descriptor with an additional version info entry.  Nodes
+  must upgrade their runtime binary and configuration by the
+  `ValidFrom` epoch or the runtime will halt.
```
Comment (kostko) [inline docs/adr/0013-runtime-upgrades.md:101]: By halt you probably mean that the runtime will be suspended?

File: docs/adr/0013-runtime-upgrades.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+- Aborting or altering pending updates via the deployment of a new version
+  of the descriptor with the removed/ammended not-yet-valid `Deployments`
+  is possible in this design, but perhaps should be forbidden.
+
+- Altering exisiting `Deployments` entries is strictly forbidden.
```
Comment (kostko) [inline docs/adr/0013-runtime-upgrades.md:107]: But removal of superseded deployments must be possible? E.g. once a deployment is valid (the current epoch is equal to or greater than `ValidFrom`) all previous ones could be removed. Or will this happen automatically?

**Comment by Yawning** [comment]
Superseded by #4449

**Pull request closed without merging**